### PR TITLE
YG-407: Refactored Attendance Report Modal.

### PR DESCRIPTION
**Pull request by pepcus-priya**
# Related issues
- V2 > Refactored Attendance Report Modal.

# Proposed changes
- used useContext and Text Components.
- Removed pepcus-core library
- Replaced Class Component to Function Component.

# Screenshots
image
image

### CODE REVIEW

File: src/v2/components/Files/fileStyled.js [hunk 1]
```diff
@@ -0,0 +1,21 @@
+import styled from 'styled-components';
+
+export const FileDownloadIcon = styled('a')`
+	cursor: pointer;
+	margin-left: 10px;
```
Comment (pepcus-mohit) [inline src/v2/components/Files/fileStyled.js:5]: Use theme.spacing

File: src/v2/components/Files/fileStyled.js [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+	color: ${(props) => (props.isview === 'true' ? '#0868ac' : '#898787')};
```
Comment (pepcus-mohit) [inline src/v2/components/Files/fileStyled.js:6]: Use theme.color instead of hard codings

File: src/v2/components/design/modal/Modal.tsx [hunk 3]
```diff
@@ -1,9 +1,54 @@
...
 };
-const StyledContainer = styled.div``;
+
+const StyledOverlayContainer = styled.div`
+	position: fixed;
+	top: 0;
+	left: 0;
+	width: 100%;
+	height: 100%;
+	background-color: rgba(0, 0, 0, 0.5);
```
Comment (pepcus-mohit) [inline src/v2/components/design/modal/Modal.tsx:42]: Use theme.color instead

File: src/v2/components/design/modal/ModalHeader.jsx [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+const ModalHeader = ({ children }) => {
+	return <ModalHeaderContainer>{children}</ModalHeaderContainer>;
+};
+
+const ModalHeaderContainer = styled.div`
+	box-sizing: border-box;
+	flex: 0 1 auto;
+	flex-flow: row wrap;
+	width: auto;
+	background: rgb(88, 100, 251);
```
Comment (pepcus-mohit) [inline src/v2/components/design/modal/ModalHeader.jsx:13]: Use theme.color

File: src/v2/components/loader/CustomLoader.jsx [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+ */
+
+const LoaderStyled = styled.div`
+	background: #0868ac;
+`;
+
+const CustomLoader = () => (
+	<StyledLoaderContainer>
+		<StyledLoader>
+			<div className="lds-css">
```
Comment (pepcus-mohit) [inline src/v2/components/loader/CustomLoader.jsx:18]: No classnames, Use StyledComponents only

File: src/v2/layouts/AttendanceReportModal/AttendanceReportModal.jsx [hunk 6]
```diff
@@ -0,0 +1,259 @@
...
+				const classRooms = () => {
+					return Object.keys(value.classRooms)?.map((classRoom) => ({
+						classroom: classRoom,
+						classroomValues: value.classRooms[classRoom],
+					}));
+				};
+				formattedData.push({
+					day: key.toString(),
+					...value,
+					classRooms: classRooms(),
```
Comment (pepcus-mohit) [inline src/v2/layouts/AttendanceReportModal/AttendanceReportModal.jsx:52]:
```suggestion
					classRooms,
```

File: src/v2/layouts/AttendanceReportModal/AttendanceReportModal.jsx [hunk 7]
```diff
@@ -0,0 +1,259 @@
+import { useContext, useEffect, useState } from 'react';
```
Comment (pepcus-mohit) [inline src/v2/layouts/AttendanceReportModal/AttendanceReportModal.jsx:1]: This component is too long, Try to accommodate the component's content in 100 lines of code.

File: src/v2/layouts/AttendanceReportModal/AttendanceReportModal.jsx [hunk 8]
```diff
@@ -0,0 +1,259 @@
...
+	const closeModal = () => {
+		setIsModalOpen(false);
+	};
+
+	const renderModal = () => {
+		if (isModalOpen) {
+			return (
+				<Modal closeModal={closeModal} isShowing={isModalOpen}>
+					<StyledModalContent>
+						<ModalHeader>
```
Comment (pepcus-mohit) [inline src/v2/layouts/AttendanceReportModal/AttendanceReportModal.jsx:104]: We can move the Modal Content part in separate component to shorten the component's length.

File: src/v2/layouts/AttendanceReportModal/AttendanceReportModal.jsx [hunk 9]
```diff
@@ -0,0 +1,259 @@
...
+							<StyledModelHeaderContent>
+								<StyledHeaderTitle>
```
Comment (pepcus-mohit) [inline src/v2/layouts/AttendanceReportModal/AttendanceReportModal.jsx:106]: Try to use TextComponents wherever we're displaying texts

File: src/v2/layouts/AttendanceReportModal/AttendanceReportModal.jsx [hunk 10]
```diff
@@ -0,0 +1,259 @@
...
+	margin: -1px -1px 20px;
+	background: rgb(88, 100, 251);
+	-webkit-box-pack: center;
+	justify-content: center;
+	display: block;
+`;
+
+const StyledHeaderTitle = styled.div`
+	font-family: Poppins, sans-serif;
+	color: rgb(255, 255, 255);
```
Comment (pepcus-mohit) [inline src/v2/layouts/AttendanceReportModal/AttendanceReportModal.jsx:177]: Use theme.color

File: src/v2/layouts/AttendanceReportModal/AttendanceReportModal.jsx [hunk 11]
```diff
@@ -0,0 +1,259 @@
...
+						</ModalHeader>
+						<StyledModalBody>
+							{attendanceFormattedResponse?.map((attendanceResponseObj) => {
+								return (
+									<Accordion
+										key={attendanceResponseObj.day}
+										title={
+											<StyledTitleContainer>
+												{attendanceResponseObj.day}{' '}
+												<StyledSpan>
```
Comment (pepcus-mohit) [inline src/v2/layouts/AttendanceReportModal/AttendanceReportModal.jsx:122]: Use Text Components instead

File: src/v2/layouts/AttendanceReportModal/AttendanceReportModal.jsx [hunk 12]
```diff
@@ -0,0 +1,259 @@
...
+	justify-content: space-between;
+	position: relative;
+	padding-right: 20px;
+
+	a:before {
+		content: '';
+		position: absolute;
+		width: 17px;
+		height: 41px;
+		border-right: 1px solid #e5e5e5;
```
Comment (pepcus-mohit) [inline src/v2/layouts/AttendanceReportModal/AttendanceReportModal.jsx:232]: Use theme.color

File: src/v2/layouts/AttendanceReportModal/AttendanceReportModal.jsx [hunk 13]
```diff
@@ -0,0 +1,259 @@
...
+	&::-webkit-scrollbar {
+		width: 5px;
+	}
+
+	&::-webkit-scrollbar-track {
+		background: transparent;
+	}
+
+	&::-webkit-scrollbar-thumb {
+		background: #888;
```
Comment (pepcus-mohit) [inline src/v2/layouts/AttendanceReportModal/AttendanceReportModal.jsx:251]: Use theme.color

File: src/v2/layouts/AttendanceReportModal/AttendanceReportModal.jsx [hunk 14]
```diff
@@ -0,0 +1,259 @@
...
+		width: 17px;
+		height: 41px;
+		border-right: 1px solid #e5e5e5;
+		top: -10px;
+		right: 8px;
+	}
+`;
+
+const StyledModalBody = styled.div`
+	max-height: 76vh;
```
Comment (pepcus-mohit) [inline src/v2/layouts/AttendanceReportModal/AttendanceReportModal.jsx:239]: Keep consistent units, it should be px throughout the app.

File: src/v2/layouts/MemberDataGrid/MemberInfoData.tsx [hunk 15]
```diff
@@ -18,18 +17,22 @@ import { GridHeaderType } from './memberInfoDataTypes';
...
 	const { response } = useAPI(memberInfoGridViewApi({ secretKey: secretKey }));
-	const { students = [] } = response || {};
-	const [studentList, setStudentList] = useState<StudentCorrectionResponse[]>([]);
+	const { studentList = [] } = response || {};
+	const [studentListData, setStudentListData] = useState<StudentCorrectionResponse[]>([]);
 	const [checkIds, setCheckIds] = useState<StudentCorrectionResponse[]>([]);
 	const [isVisibleConfigModal, setIsVisibleConfigModal] = useState<boolean>(false);
 	const { headerConfig } = memberGridMetaData;
 
+	console.log('####response', studentList);
```
Comment (pepcus-mohit) [inline src/v2/layouts/MemberDataGrid/MemberInfoData.tsx:34]: Remove console.

File: src/v2/layouts/MemberDataGrid/MemberInfoData.tsx [hunk 16]
```diff
@@ -8,7 +8,6 @@ import { useAPI } from '../../components/hooks';
 import isEmpty from 'lodash/isEmpty';
```
Comment (pepcus-mohit) [inline src/v2/layouts/MemberDataGrid/MemberInfoData.tsx:8]: Break this component into multiple components, A component should not exceed 100-125 lines

File: src/v2/layouts/MemberDataGrid/MemberInfoData.tsx [hunk 17]
```diff
@@ -155,6 +155,7 @@ export const MemberInfoData = (): JSX.Element => {
 				<MemberConfigSearch onSearch={onSearch} clearSearch={clearSearch} />
```
Comment (pepcus-mohit) [inline src/v2/layouts/MemberDataGrid/MemberInfoData.tsx:155]: Move Search functionality to another component

File: src/v2/layouts/MemberDataGrid/MemberInfoData.tsx [hunk 18]
```diff
@@ -164,7 +165,7 @@ export const MemberInfoData = (): JSX.Element => {
 				) : (
 					<MemberInfoGridView
 						metaData={metaData}
-						students={studentList.length ? studentList : students}
+						students={studentListData.length ? studentListData : studentList}
 						getSelectedRow={getSelectedRow}
 					/>
 				)}
```
Comment (pepcus-mohit) [inline src/v2/layouts/MemberDataGrid/MemberInfoData.tsx:171]: Also move the column config handling in another component.

File: src/v2/layouts/MemberRegistrationCorrection/MemberCorrectionResponseModal.tsx [hunk 19]
```diff
@@ -1,7 +1,7 @@
-import { useContext } from 'react';
+import { useContext, useState } from 'react';
 import { StringsContext } from '../../components/providers/StringsProvider/StringsContext';
 import styled, { css } from 'styled-components';
-import { Modal } from '../../components/design/modal/Modal';
+import Modal from '../../components/design/modal/Modal';
```
Comment (pepcus-mohit) [inline src/v2/layouts/MemberRegistrationCorrection/MemberCorrectionResponseModal.tsx:4]:
```suggestion
import { Modal } from '../../components/design/modal/Modal';
```

File: src/v2/layouts/SelectedMembersWrapper/SelectedMembersWrapper.jsx [hunk 20]
```diff
@@ -0,0 +1,100 @@
...
+			</StyledDropdownButton>
+			<StyledDropdownContent>
+				<AttendanceReportModal />
+			</StyledDropdownContent>
+		</StyledDropdownContainer>
+	);
+};
+
+const StyledDropdownButton = styled(Button)`
+	border: none rgb(219, 215, 214);
```
Comment (pepcus-mohit) [inline src/v2/layouts/SelectedMembersWrapper/SelectedMembersWrapper.jsx:31]: Use theme.color

File: src/v2/layouts/SelectedMembersWrapper/SelectedMembersWrapper.jsx [hunk 21]
```diff
@@ -0,0 +1,100 @@
...
+	cursor: pointer;
+	position: relative;
+	text-align: center;
+	text-decoration: none;
+	text-transform: initial;
+	transition: box-shadow 0.2s ease-in 0s;
+	background: linear-gradient(to top, rgb(235, 235, 236), rgb(235, 235, 236));
```
Comment (pepcus-mohit) [inline src/v2/layouts/SelectedMembersWrapper/SelectedMembersWrapper.jsx:38]: Use theme.color

File: src/v2/layouts/SelectedMembersWrapper/SelectedMembersWrapper.jsx [hunk 22]
```diff
@@ -0,0 +1,100 @@
...
+	border-radius: 4px;
+	color: rgb(1, 3, 6);
+	margin: 0;
+	padding: 5px 15px;
+`;
+
+const StyledDropdownContent = styled.div`
+	display: none;
+	position: absolute;
+	background-color: #f9f9f9;
```
Comment (pepcus-mohit) [inline src/v2/layouts/SelectedMembersWrapper/SelectedMembersWrapper.jsx:48]: Use theme.color

File: src/v2/layouts/SelectedMembersWrapper/SelectedMembersWrapper.jsx [hunk 23]
```diff
@@ -0,0 +1,100 @@
+import styled, { css } from 'styled-components';
+import { faAngleDown, faUser } from '@fortawesome/free-solid-svg-icons';
+import { useContext } from 'react';
+import { StringsContext } from '../../components/providers/StringsProvider/StringsContext';
+import AttendanceReportModal from '../AttendanceReportModal/AttendanceReportModal';
+import { Button, Caption } from '../../components/design/text/TextComponent';
+import { Icon } from '../../components/design/Icon/Icon';
+
+const SelectedMembersWrapper = () => {
```
Comment (pepcus-mohit) [inline src/v2/layouts/SelectedMembersWrapper/SelectedMembersWrapper.jsx:9]: Can we rename this component to more specific name?

File: src/v2/components/design/modal/Modal.tsx [hunk 24]
```diff
@@ -1,9 +1,54 @@
...
-export type ModalProps = {
+export type modalProps = {
 	children?: JSX.Element;
+	isShowing?: boolean;
+	closeModal: () => void;
 };
-export const Modal = ({ children }: ModalProps): JSX.Element => {
-	return <StyledContainer>{children}</StyledContainer>;
+
+const Modal = ({ children, isShowing, closeModal }: modalProps) => {
```
Comment (pepcus-mohit) [inline src/v2/components/design/modal/Modal.tsx:11]: isShowing prop seems redundant here. We can manage the Modal visibility from parent component itself.
```suggestion
const Modal = ({ children, onCloseModal }: modalProps) => {
```

File: src/v2/layouts/SelectedMembersWrapper/SelectedMembersWrapper.jsx [hunk 25]
```diff
@@ -0,0 +1,100 @@
...
+  margin: 0 0 0 10px;
+  pointer-events: ${(props) => (props.isEmpty === true ? 'none' : 'auto')};
+
+
+  &:hover ${StyledDropdownContent} {
+    display: block;
+  }
+
+  &:hover ${StyledDropdownButton} {
+    background: #dbd7d6;
```
Comment (pepcus-mohit) [inline src/v2/layouts/SelectedMembersWrapper/SelectedMembersWrapper.jsx:75]: Use theme.color

File: src/v2/layouts/SelectedMembersWrapper/SelectedMembersWrapper.jsx [hunk 26]
```diff
@@ -0,0 +1,100 @@
...
+
+	a {
+		color: black;
+		padding: 12px 25px;
+		text-decoration: none;
+		display: block;
+		border-bottom: 1px solid #e5e5e5;
+
+		&:hover {
+			background-color: #f1f1f1;
```
Comment (pepcus-mohit) [inline src/v2/layouts/SelectedMembersWrapper/SelectedMembersWrapper.jsx:60]: Use theme.color

**Review by pepcus-mohit — CHANGES_REQUESTED** [review]

**Comment by pepcus-ankit** [comment]
@pepcus-priya : Please try to follow coding practices. @pepcus-mohit : Could you point her to the coding practices we follow.

**Pull request merged**